compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -3250,7 +3250,14 @@ public void visit(BLangConstant constant) {
             dlog.error(constant.name.pos, DiagnosticErrorCode.UNDERSCORE_NOT_ALLOWED);
         }
         if (constant.typeNode != null && !types.isAllowedConstantType(constant.typeNode.type)) {
-            dlog.error(constant.typeNode.pos, DiagnosticErrorCode.CANNOT_DEFINE_CONSTANT_WITH_TYPE, constant.typeNode);
+            if (TypeTags.isIntegerTypeTag(constant.typeNode.type.tag) ||
+                    TypeTags.isStringTypeTag(constant.typeNode.type.tag)) {
```

Comment:
Shall we do this for all the supported types from the spec?

>  The type-descriptor must specify a type that is a subtype of anydata and must not be never.

So basically, `types.isAssignable(type, symTable.anydataType) && !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(type)`